src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,78 @@
-import React from 'react';
+import React, { ChangeEvent, useCallback, useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { getMovie } from '../../api';
+
+import { MovieCard } from '../MovieCard';
+
+type Props = {
+  addMovie: (movie: Movie) => void,
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [searcResult, setSearcResult] = useState<MovieData | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isSubmit, setIsSubmit] = useState(false);
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    getMovie(query)
+      .then((response) => {
+        if ((response as MovieData).Title) {
+          setSearcResult(response as MovieData);
+          setIsSubmit(true);
+
+          return;
+        }
+
+        setHasError(true);
+      })
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleClear = () => {
+    setQuery('');
+    setHasError(false);
+    setIsSubmit(false);
+    setSearcResult(null);
+  };
+
+  const handleQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setHasError(false);
+    setIsSubmit(false);
+    setSearcResult(null);
+  };
+
+  const getNewMovie = useCallback((newMovie) => {
+    return {
+      title: newMovie.Title,
+      description: newMovie.Plot,
+      imdbId: newMovie.imdbID,
+      imgUrl: newMovie.Poster !== 'N/A'
+        ? newMovie.Poster
+        : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+      imdbUrl: `https://www.imdb.com/title/${newMovie.imdbID}`,
+    };
+  }, [searcResult]);
```

Comment:
guess it's supposed to be `searchResult` )
```suggestion
  }, [searchResult]);
```